src/Variable.cpp

```diff
@@ -35,7 +35,7 @@ Variable::Variable(const char *name)
 // Boolean or String
 Variable::Variable(const char *name, const char *value, const char *helpstr, int flags, FnChangeCallback_t callback)
 	: Variable() {
-	if (flags != 0)
+	if (flags & ~FCVAR_DONTRECORD != 0)
```

Comment:
Precedence bug! This groups as `flags & (~FCVAR_DONTRECORD != 0)`. Change to `if ((flags & ~FCVAR_DONTRECORD) != 0)`